pkg/expr/threshold.go

```diff
@@ -91,6 +91,17 @@ func (tc *ThresholdCommand) NeedsVars() []string {
 }
 
 func (tc *ThresholdCommand) Execute(ctx context.Context, now time.Time, vars mathexp.Vars, tracer tracing.Tracer) (mathexp.Results, error) {
+	switch tc.ThresholdFunc {
+	case ThresholdIsOutsideRange, ThresholdIsWithinRange:
+		if len(tc.Conditions) < 2 {
+			return mathexp.Results{}, fmt.Errorf("failed to evaluate threshold expression: incorrect number of arguments: got %d but need 2", len(tc.Conditions))
+		}
+	case ThresholdIsAbove, ThresholdIsBelow:
+		if len(tc.Conditions) < 1 {
+			return mathexp.Results{}, fmt.Errorf("failed to evaluate threshold expression: incorrect number of arguments: got %d but need 1", len(tc.Conditions))
+		}
+	}
```

Comment:
I think this the better place for this logic is in `createMathExpression` closer to the construction of math expression.